### ISSUE TITLE
torch.chunk docs seem underspecified

### ISSUE THREAD

**Issue by saketh-are** [pytorch/pytorch#59983]
## 📚 Documentation

[torch.chunk](https://pytorch.org/docs/stable/generated/torch.chunk.html)'s documentation doesn't fully specify how the number and sizes of chunks produced are chosen.

Particularly, it's not clear what happens in cases where the size of the given dimension is not divisible by the number of chunks requested. One of the parameters to the function is "chunks (int) – number of chunks to return," but torch.chunk may not actually return that many chunks. For example,

```
>>> torch.chunk(torch.tensor([1, 2, 3, 4, 5]), chunks=4)
(tensor([1, 2]), tensor([3, 4]), tensor([5]))
```

The page could also benefit from some usage examples.

cc @brianjo @mruberry

**Comment by mruberry** [comment]
It's always great to have better docs.

**Comment by dagitses** [comment]
fixed by #61097
